### ISSUE TITLE
No imports found

### ISSUE THREAD

**Issue by Zyets** [unlight/sublime-import-helper#65]
ImportHelper at some point lost all source modules and refuses to update them.

Here is log from debug mode:
```
source_folders: ['/home/zyets/projects/client']
update_node_modules: import_root: /home/zyets/projects/client
typescript_paths: []
Run command: ['/usr/local/bin/node', 'get_folders', {'folders': ['/home/zyets/projects/client']}]
Run command: ['/usr/local/bin/node', 'get_modules', {'importRoot': '/home/zyets/projects/client', 'packageKeys': ['devDependencies']}]Run command: ['/usr/local/bin/node', 'get_modules', {'importRoot': '/home/zyets/projects/client', 'packageKeys': ['dependencies']}]Run command: ['/usr/local/bin/node', 'get_from_package', {'importRoot': '/home/zyets/projects/client'}]

get_from_package_callback: node_modules_names: {'enzyme-adapter-react-16', '@types/react-datepicker', 'react-dom', '@types/react-dom', '@types/react-custom-scrollbars', 'react-router-dom', 'raf', '@types/react-redux', 'source-map-loader', '@turf/turf', 'postcss-load-config', 'react-ace', 'retyped-turf-tsd-ambient', '@types/rc-slider', 'babel-eslint', 'whatwg-fetch', '@types/redux-logger', 'rimraf', 'eslint-loader', 'sockjs-client', '@types/redux-actions', '@types/react', 'typescript', '@types/stompjs', 'style-loader', 'react-redux-i18n', '@types/proj4', 'react-test-renderer', 'babel-polyfill', 'react-redux', 'babel-core', 'font-awesome', 'react', 'fetch-mock', 'pretty-quick', 'eventsource-polyfill', 'openlayers', 'react-addons-test-utils', 'webpack-svgstore-plugin', '@types/sockjs-client', 'eslint-plugin-react', 'shelljs', 'babel-runtime', 'copy-webpack-plugin', 'babel-jest', 'redux-logger', '@types/enzyme', 'enzyme', 'react-datetime', 'isomorphic-fetch', 'rc-slider', 'precss', 'postcss-loader', 'css-loader', 'ts-jest', 'babel-loader', 'react-player', 'resize-observer-polyfill', 'react-custom-scrollbars', 'react-markdown', 'ts-loader', 'regenerator-runtime', 'redux-mock-store', 'classnames', 'moment', 'webpack-cli', 'html-webpack-plugin', 'raw-body', '@types/react-router-dom', 'tslint-config-prettier', '@types/react-router', 'url-loader', 'd3', 'react-table', 'prop-types', 'file-loader', '@types/d3', 'redux-actions', '@types/react-dropzone', 'load-js', 'babel-plugin-react-intl', 'react-dropzone', 'postcss-import', '@types/react-redux-i18n', 'json-loader', 'stompjs', 'identity-obj-proxy', 'babel-plugin-transform-class-properties', 'babel-preset-react', '@types/classnames', 'webpack-dev-server', 'react-input-mask', 'cross-env', 'babel-preset-es2015', '@types/geojson', 'autoprefixer', 'husky', 'redux-thunk', 'moment-timezone', '@types/redux-mock-store', '@types/draft-js', 'typescript-babel-jest', 'ora', 'react-router-redux', 'postcss-cssnext', 'webpack-bundle-analyzer', 'history', '@types/jest', 'babel-plugin-transform-es2015-modules-commonjs', 'draft-js', 'react-autosize-textarea', 'react-moment', 'redux', '@types/history', 'jest', 'cerialize', 'webpack-node-externals', '@types/lodash', '@types/react-router-redux', '@types/node', 'webpack', 'tslint', 'bundle-loader', 'sinon', 'eslint', '@types/moment-timezone', 'lodash', 'react-css-modules', 'babel-plugin-transform-object-rest-spread', 'xml-json-parser', 'redux-saga', 'less', 'babel-plugin-transform-runtime', 'prettier', 'keycloak-js', '@types/fetch-mock', 'react-router', 'react-datepicker', '@types/openlayers', 'geojson', 'x2js', 'omggif', 'es6-shim', '@types/react-table', 'tslint-loader', 'proj4', 'lodash.iserror', 'less-loader'}
get_modules_callback: len(result): 4417
get_modules_callback: len(result): 4946
Update source modules: 0
```

I've got package file and its dumb - only project path, I removed all other settings and exclusions. Still no source modules.

I went into package source code and found out that get_folder script returns empty array to this method (result is []):
> def get_source_modules_callback(err, result)

I dont know how to debug get_folder.js script, so can not provide more info on that issue.

Also my env info:
Sublime v3.2.1 Build 3207
ImportHelper v2.1.0
node v8.9.1
npm v6.9.0

Please, help me.

**Comment by unlight** [comment]
Please, provide contents of your project file `.sublime-project`

**Comment by Zyets** [comment]
{ "folders": [ { "path": "/home/zyets/projects/client" }, ] }

**Comment by Zyets** [comment]
Also, i've already tried to reinstall the package few times but that did not help at all.

**Comment by unlight** [comment]
Ok, let's try to figure out the problem.
1. Goto sublime_text folder -> Data -> Packages -> Import Helper
2. Run `node backend_run.js get_folders '{"folders":["/home/zyets/projects/client"]}'`

You should see empty array.
It is not expected, because in this folder your source code.

Try to create a new folder e.g. `client2` and create js file with content:
```
export class X { }
```
Then run `node backend_run.js get_folders '{"folders":["/home/zyets/projects/client2"]}'`
If you see non empty result, it means get_folders command is working ok, the issue in your source code, try to move folders one by one to new `client2` folder and run `node backend_run.js get_folders '{"folders":["/home/zyets/projects/client2"]}'` after each move.

When you see empty result, last moved folder is the problem one.
Repeat same steps for that folder if nesessary.

You can use bisection method for faster digging (moving not one by one folder, but half of previous on each move)

**Comment by Zyets** [comment]
Thanks a lot for reply! It helped!
I found out following strange things:

The code was fine, but compiled app was crashing the process. To be precise, vendor.bundle.js was doing that, but i dont think problem is actually with the file. I was looking for exact spot in the code which makes troubles, so i cut chunk by chunk out of the code, until the command started returning correct result, and so i ended up with this: [vendor.bundle.js](https://www.dropbox.com/s/4yc8gjlx6q6i6t5/vendor.bundle.js?dl=0)
I guess its a part of moment-timezone package.

Interesting thing is that if you delete ANY of 1187 line, everything works fine.

I guess problem lies in a esm-exports package but i did not check that yet.

**Comment by unlight** [comment]
@Zyets 
You should exclude files such `vendor.bundle.js` or any other binaries from source folder and prevent parsing it.

Can you please provide the scope - what line crashed `esm-exports`?

**Comment by Zyets** [comment]
Ok. Here's my initial project file:

``` JSON
{
	"folders":
	[
		{
			"file_exclude_patterns":
			[
				"report.html"
			],
			"folder_exclude_patterns":
			[
				"dist",
				"node_modules",
				"coverage",
				"server",
				"typings",
				"target",
				"client/geoportal/dist",
				"client/geoportal/node_modules",
				"client/geoportal/plugins",
				"report.html"
			],
			"path": "/home/zyets/projects/client"
		},
	]
}
```
There are few directories, which may contain bundle files. Normally, they are all excluded from the project.

Also I ran through `get_source_folders` (utils.py) method, it ignores all the `exclude_patterns`. So does the get_folders command.

I think that should be fixed in future updates, but not critical. Should i reopen the issue?

Also i can not tell you which line crashes esm-exports, cause it doesn't crash at all. It returns empty array, with no errors, log is clear, as i showed you before (1st message).

**Comment by unlight** [comment]
> It returns empty array, with no errors, log is clear, as i showed you before (1st message).

Ok, it is because esm-exports fails silently and returns empty array.

There is an issue when `exclude` in plugin happens after get_source_command
This is already fixed in master branch
https://github.com/unlight/sublime-import-helper/pull/64

Will released in v2.2.0 today evening.
